File: ShadowFrameworkLite20/src/shadow/geometry/curves/data/SFContinuosBezier3Data.java
package shadow.geometry.curves.data;

import shadow.geometry.SFValuesList;
import shadow.geometry.curves.SFBezier3;
import shadow.geometry.curves.SFControlPointsCurve;
import shadow.geometry.curves.SFUniformCurvesSpline;
import shadow.math.SFValuenf;
import shadow.math.SFVertex3f;

public class SFContinuosBezier3Data extends SFCurvesVerticesData{

	protected SFControlPointsCurve buildResource() {
		
		SFUniformCurvesSpline<SFValuenf,SFBezier3<SFValuenf>>  spline
				=new SFUniformCurvesSpline<SFValuenf, SFBezier3<SFValuenf>>();
		
		SFValuesList<SFValuenf> vertices=this.vertices.getDataset().getResource();
		
		for (int i = 0; i < vertices.getSize(); i++) {
			SFVertex3f point=new SFVertex3f();
			vertices.getValue(i, point);
			spline.addControlPoint(point);
		}
		
		for (int i = 0; i < vertices.getSize()-3; i+=3) {

			SFValuenf A=spline.getControlPoint(i);
			SFValuenf B=spline.getControlPoint(i+1);
			SFValuenf C=spline.getControlPoint(i+2);
			SFValuenf D=spline.getControlPoint(i+3);

			SFBezier3<SFValuenf> values=new SFBezier3<SFValuenf>(A, B, C, D);
			spline.addCurve(values);
		}
		
		return spline;
	}
}

File: ShadowFrameworkJS/src/shadow/geometry/curves/data/SFContinuosBezier3Data.js




function SFContinuosBezier3Data(){
	this.vertices = new SFDataAssetObject(null);
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.vertices);
	this.setData(parameters);
}

inherit(SFContinuosBezier3Data,SFCurvesVerticesData);

SFContinuosBezier3Data.prototype["buildResource"]=function(){
	
	var  spline=new SFUniformCurvesSpline();
	
	var vertices=this.vertices.getDataset().getResource();
	
	for (var i = 0; i < vertices.getSize(); i++) {
		var point=new SFVertex3f();
		vertices.getValue(i, point);
		spline.addControlPoint(point);
	}
	
	for (var i = 0; i < vertices.getSize()-3; i+=3) {

		var A=spline.getControlPoint(i);
		var B=spline.getControlPoint(i+1);
		var C=spline.getControlPoint(i+2);
		var D=spline.getControlPoint(i+3);

		var values=new SFBezier3(A, B, C, D);
		spline.addCurve(values);
	}
	
	return spline;
};

SFContinuosBezier3Data.prototype["generateNewDatasetInstance"]=function(){
	return new SFContinuosBezier3Data();
};	



